core-bundle/src/EventListener/FinalizeResponseContextListener.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContextAccessor;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+
+class FinalizeResponseContextListener
+{
+    public function __construct(private readonly ResponseContextAccessor $responseContextAccessor)
+    {
+    }
+
+    public function __invoke(ResponseEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $this->responseContextAccessor->finalizeCurrentContext($event->getResponse());
```

Comment:
Not sure if I like this. It was designed to be called by the controller in order to make sure, the control over this remains in the controller. Now, a controller cannot control this anymore :/ Why exactly did you have to move it here?

Reply:
I don't like it either. See the comment here: https://github.com/contao/contao/pull/6631/files/99ac12a622fc62d6d94a0943f143f2ebdfc10eef#diff-1ed02a947ba36ca7331ab438573363a620fc31d0ccd3f7e09c4c5b46e5574d54

We can work around it though. For example our `PreviewToolbarListener` could generate their own nonces and add them to the CSP headers if present, just like Symfony Profiler Event Listener does. Our `PreviewToolbarListener` could even use its own `CspHandler` instance for that. If we agree on this I'll implement it this way.

Reply:
Changed in 0ceea3f50d